### ISSUE TITLE
Clarification on how the training/test split is defined and used

### ISSUE THREAD

**Issue by HaoTao-824** [ziangcao0312/PhysX-Anything#31]
Hi, thank you for open-sourcing this excellent project.

  I am trying to better understand how the training and test sets are defined in PhysX-Anything, especially for reproducing the training/evaluation pipeline.

  From reading README.md and the scripts in dataset/, I noticed the following:

  - The repository includes trainingset.npy and testset.npy, which seem to define an object-level split.
  - trainingset.npy and testset.npy appear to be disjoint.
  - However, in the current preprocessing/data-generation scripts, especially:
      - dataset/2encode_representation_32_finetune_mobility.py
      - dataset/3generate_data_new_32_finetune_mobility.py

    it seems that all objects under the processed directories may be converted into training data, rather than being explicitly filtered by trainingset.npy.
  - In addition, evaluation_phy.py loads val_test_list.npy, but this file is not included in the repository, so I am not sure how it relates to testset.npy.

  Could you please clarify the following points?

  1. How exactly are the training set and test set split?
      - Does it apply only to PhysX-Mobility, or also to PhysXNet?
  2. During training data generation, should trainingset.npy be used to filter the objects explicitly?
      - If yes, which script is expected to enforce this filtering?
  3. What is the relationship between testset.npy and val_test_list.npy?
      - Are they the same split, or is one a subset of the other?
  4. If possible, could you share the intended protocol for reproducing the official train/test split?

**Comment by ziangcao0312** [comment]
1. The training and test sets were selected by random sampling with NumPy. Specifically, val_test_list is a test set randomly sampled from PhysXNet, while testset is a test set randomly sampled from PhysX-Mobility.
 
2. When processing the data, we considered that some researchers might prefer to define their own training and test splits. Therefore, our code does not directly load the specific training and test sets we selected. If you would like to align with our setting, you only need to check during data processing whether each sample belongs to the training set.

3. These two sets were both selected randomly. Although the sampling procedure is the same, they were sampled independently, so neither one is a subset of the other.

4. We use the numpy to select the testset:

selected = np.random.choice(os.listdir("dataset_path"), size=n, replace=False)

**Comment by HaoTao-824** [comment]
Thanks a lot!!!!  Clear instructions

**Comment by HaoTao-824** [comment]
I have another question regarding the dataset splits.

While trainset.npy and testset.npy in the repository provide the splits for physx-mobility, I am not entirely sure how the dataset was split for physx-net,because I couldn't find the val_test_list file in the repository.. Could you clarify how the training set was divided for it? Or was the entire dataset used for training?
